src/main/java/build/buildfarm/instance/shard/ShardInstance.java

```diff
@@ -601,11 +601,21 @@ private void findMissingBlobsOnWorker(
       Executor executor,
       SettableFuture<Iterable<Digest>> missingDigestsFuture,
       RequestMetadata requestMetadata) {
-    String worker = workers.removeFirst();
+    Instance workerStub = null;
+    String worker;
+    do {
+      worker = workers.removeFirst();
+      try {
+        workerStub = workerStub(worker);
```

Comment:
the obvious name collision here is a problem, and in general, I think that we should be making the backplane responsible for the validity of these contents - it should never return an unparseable worker from its workers presentation.